bfcfg

```diff
@@ -648,29 +655,157 @@ sys_cfg_one_byte()
   fi
 }
 
+#
+# Dump a multi-byte value at the specified offset in a file.
+#
+sys_cfg_multi_byte()
+{
+  local tmp_file=$1
+  local name=$2
+  local offset=$3
+  local num_bytes=$4
+  local value=$5
+
+  if [ ${dump_mode} -eq 0 ]; then
+    return
+  fi
+
+  value=0
+  if [ -e ${tmp_file} ]; then
+    value=$(hexdump -s "${offset}" -n "${num_bytes}" -e '"0x%X" "\n"' "${tmp_file}")
+  fi
+  echo "sys: ${name}=${value}"
+}
+
+#
+# This function reads the BfModes EFI struct from a UEFI variable directly.
+# Below are the offsets defined in UEFI which is 4(fixed header) plus the offset
+# of the variable within the BfModes struct. These offsets are not supposed to
+# change in order to be backward compatible with previous releases.
+#   VARIABLE(Name)       OFFSET(Byte)  SIZE(Byte)
+#   INTERNAL_CPU_MODEL        4            1
+#   HOST_PRIV_LEVEL           5            1
+#   NIC_MODE                  6            1
+#
+sys_cfg_dump_bf_modes()
+{
+  local tmp_file=/tmp/.bfcfg-sysfs-data
```

Comment:
1. same comment here to use a different name.
2. Would it be possible to use a temp directory (check example in bfver line 31-33), and replace all /tmp/xxx with ${TMP_DIR}/xxx?  So it could avoid race condition when multiple people run it is parallel.

Reply:
The only thing I had to revert just now was the log file located at /tmp/bfcfg.log since that file has to persist after the script exits. All other instances of /tmp/X are using TMP_DIR now though